hack/coredump.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Call this to dump all master and node logs into the folder specified in $1
+# (default _artifacts). Only works if the provider supports SSH.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Saves logs common to master and nodes. The node name is in $1 and the
+# directory/name prefix is in $2.
+function cat-logs() {
+    local node_name="${1}"
+    local prefix="${2}"
+    echo "Dumping logs for ${node_name}"
+    if ssh-to-node "${node_name}" "sudo systemctl status kubelet.service" &> /dev/null; then
+        ssh-to-node "${node_name}" "sudo journalctl --output=cat -u kubelet.service" > "${prefix}-kubelet.log"
+    else
+        ssh-to-node "${node_name}" "cat /var/log/kubelet.log" > "${prefix}-kubelet.log"
+        ssh-to-node "${node_name}" "cat /var/log/supervisor/supervisord.log" > "${prefix}-supervisord.log"
+        ssh-to-node "${node_name}" "cat /var/log/supervisor/kubelet-stdout.log" > "${prefix}-supervisord-kubelet-stdout.log"
+        ssh-to-node "${node_name}" "cat /var/log/supervisor/kubelet-stderr.log" > "${prefix}-supervisord-kubelet-stderr.log"
+        ssh-to-node "${node_name}" "cat /var/log/kern.log" > "${prefix}-kern.log"
+        ssh-to-node "${node_name}" "cat /var/log/docker.log" > "${prefix}-docker.log"
+    fi
+}
+
+readonly report_dir="${1:-_artifacts}"
+echo "Dumping artifacts to ${report_dir}"
+
+: ${KUBE_CONFIG_FILE:="config-test.sh"}
+
+source "cluster/kube-env.sh"
```

Comment:
based on most of the scripts I've seen around in places like `cluster/`, I think it's 

```bash
set - o stuff

KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
source "${KUBE_ROOT}/stuff"
```